.github/workflows/CD.yml

```diff
@@ -0,0 +1,116 @@
+name: CD
+
+on:
+  push:
+    tags:        
+      - '**' 
+
+jobs:
+  validate-tag:
+    name: Validate tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Check no jumps in tags
+        run: |
+          # If a previous tag is X.Y, valid tags are only X.(Y+1) or (X+1).0
+          if [[ $(git tag -l | wc -l) != 1 ]]; then # If there is only one tag, no need to check
+            prev=($(git tag -l --sort=-v:refname | head -n 2 | tail -n 1 | tr "." " "))
+            val1="${prev[0]}.$((${prev[1]}+1))" # X.(Y+1)
+            val2="$((${prev[0]}+1)).0" # (X+1).0
+            if [[ "${{github.ref_name}}" != "$val1" ]] && \
+            [[ "${{github.ref_name}}" != "$val2" ]]; then
+                echo "Invalid tag \"${{github.ref_name}}\" when previous tag is 
+                \"$(sed 's/ /./' <<< ${prev[@]})\"."
+                exit 1
+            fi
+          fi
+  
+  get-package-name:
+    name: Get package name
+    runs-on: ubuntu-latest
+    needs: validate-tag
+    outputs:
+      package-name: ${{ steps.get-package-name.outputs.package-name }}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Get name
+        id: get-package-name
+        run: |
+          echo "package-name=$(yq -oy '.project.name' pyproject.toml)" >> $GITHUB_OUTPUT
+  
+  release-conda-package:
+    name: Build with conda and release
+    runs-on: ubuntu-latest
+    needs: [get-package-name, validate-tag]
+    env:
+      BUILD_FOLDER: ${{github.workspace}}/build
+      PACKAGE_PATH: ${{github.workspace}}/build/noarch/*.tar.bz2
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+            fetch-tags: true
+            fetch-depth: 0 
+
+      - name: Setup conda build environment
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
+        with:
+          miniconda-version: "latest"
+          python-version: ${{ vars.PY_VERSION }} 
+          environment-file: .conda/env_build.yml
+          auto-activate-base: false
+          auto-update-conda: false
+          show-channel-urls: true
+      
+      - name: Build conda package
+        shell: bash -el {0}
+        run: |
+          conda build . --no-anaconda-upload --output-folder=${{env.BUILD_FOLDER}} -c conda-forge -c accessnri -c coecms
+      
+      - name: Upload conda package to Anaconda.org
+        shell: bash -el {0}
+        run: |
+          anaconda -t ${{ secrets.ANACONDA_TOKEN }} upload --user accessnri ${{env.PACKAGE_PATH}}
```

Comment:
I would keep the user a `secret`. Just to be safe.

Reply:
Alright. 
Do you think it would work better as an Organization secret? 
Something like `ACCESS_NRI_ANACONDA_USER`?

Reply:
And yeah something like `ANACONDA_USER_NAME`, probably don't have to put `ACCESS_NRI_` at the front.

Reply:
Fixed